surface/include/pcl/surface/advancing_front_utils.h

```diff
@@ -0,0 +1,310 @@
+#ifndef PCL_SURFACE_ADVANCING_FRONT_UTILS_H_
+#define PCL_SURFACE_ADVANCING_FRONT_UTILS_H_
+
+#include <pcl/surface/advancing_front_point_type.h>
+#include <pcl/point_types.h>
+
+namespace pcl
+{
+  namespace afront
+  {
+
+    /** \brief Align pn's normal with av so they point in the same direction */
+    bool
+    alignNormal (Eigen::Vector3f &pn, const Eigen::Vector3f &av)
+    {
+      double dot = pn.dot (av);
+      if (dot < 0.0)
+      {
+        pn *= -1.0;
+        return true;
+      }
+
+      return false;
+    }
+
+    /** \brief Align pn's normal with av so they point in the same direction */
+    bool
+    alignNormal (AfrontVertexPointType &pn, const AfrontVertexPointType &av)
+    {
+      Eigen::Vector3f normal = pn.getNormalVector3fMap ();
```

Comment:
I think it should work if you pass `pn.getNormalVector3fMap ()` directly to the `alignNormal` function, the underlying data should be modified.

I'd move these two functions and the next one (`checkNormal`) to the `common` module. I don't really see any existing header file there that would naturally fit them, so propose to create a new one "normals.h".

While doing so, I'd make `alignNormals` templated on the point types, something like:

```cpp
template <typename PointNT1, typename PointNT2>
alignNormals (PointNT1& pn, const PointNT2& av)
```

to ensure reusability.